specification/src/main/asciidoc/platform/cdi-ee-spec/implementation_ee.adoc

```diff
@@ -0,0 +1,333 @@
+----
+
+The container is not required to support resources with scope other than `@Dependent`. Portable applications should not define resources with any scope other than `@Dependent`.
+
+A resource may not have a bean name.
+
+[[declaring_resource]]
+
+==== Declaring a resource
+
+A resource may be declared by specifying a Jakarta EE component environment injection annotation as part of a producer field declaration.
+The producer field may be static.
+
+* For a Jakarta EE resource, `@Resource` must be specified.
+* For a persistence context, `@PersistenceContext` must be specified.
+* For a persistence unit, `@PersistenceUnit` must be specified.
+* For a remote EJB, `@EJB` must be specified.
+* For a web service, `@WebServiceRef` must be specified.
+
+
+The injection annotation specifies the metadata needed to obtain the resource, entity manager, entity manager factory, remote EJB instance or web service reference from the component environment.
+
+[source, java]
+----
+@Produces @WebServiceRef(lookup="java:app/service/PaymentService")
+PaymentService paymentService;
+----
+
+[source, java]
+----
+@Produces @EJB(ejbLink="../their.jar#PaymentService")
+PaymentService paymentService;
+----
+
+[source, java]
+----
+@Produces @Resource(lookup="java:global/env/jdbc/CustomerDatasource")
+@CustomerDatabase Datasource customerDatabase;
+----
+
+[source, java]
+----
+@Produces @PersistenceContext(unitName="CustomerDatabase")
+@CustomerDatabase EntityManager customerDatabasePersistenceContext;
+----
+
+[source, java]
+----
+@Produces @PersistenceUnit(unitName="CustomerDatabase")
+@CustomerDatabase EntityManagerFactory customerDatabasePersistenceUnit;
+----
+
+The bean type and qualifiers of the resource are determined by the producer field declaration.
+
+If the producer field declaration specifies a bean name, the container automatically detects the problem and treats it as a definition error.
+
+If the matching object in the Jakarta EE component environment is not of the same type as the producer field declaration, the container automatically detects the problem and treats it as a definition error.
+
+[[resource_types]]
+
+==== Bean types of a resource
+
+The unrestricted set of bean types for a resource is determined by the declared type of the producer field, as specified by <<producer_field_types>>.
+
+The resulting set of bean types for a resource consists only of <<legal_bean_types,legal bean types>>, all other types are removed from the set of bean types.
+
+
+[[additional_builtin_beans]]
+
+=== Additional built-in beans
+
+A Jakarta EE or embeddable EJB container must provide the following built-in beans, all of which have qualifier `@Default`:
+
+* a bean with bean type `jakarta.transaction.UserTransaction`, allowing injection of a reference to the JTA `UserTransaction`, and
+* a bean with bean type `java.security.Principal`, allowing injection of a `Principal` representing the current caller identity.
```

Comment:
This needs to be removed here, and added to the Security spec document

Reply:
Move to: https://github.com/jakartaee/security/tree/master/spec/src/main/asciidoc

Reply:
Fixed in d56d04fc8f91b5695a05afa3bc0803e9fe2d78c8 and https://github.com/jakartaee/security/pull/323 .